keepsorted/options.go

```diff
@@ -186,45 +139,50 @@ func key(f reflect.StructField) string {
 	return key
 }
 
-func parseValue(f reflect.StructField, key, val string) (reflect.Value, error) {
-	switch f.Type {
-	case reflect.TypeFor[bool]():
-		b, ok := boolValues[val]
+func parseBlockOptions(commentMarker, options string, defaults blockOptions) (blockOptions, error) {
+	ret := defaults
+	opts := reflect.ValueOf(&ret).Elem()
+	var errs []error
+	parser := &parser{options}
+	for {
+		key, ok := parser.popKey()
 		if !ok {
-			return reflect.Zero(f.Type), fmt.Errorf("option %q has unknown value %q", key, val)
+			break
 		}
-
-		return reflect.ValueOf(b), nil
-	case reflect.TypeFor[[]string]():
-		if val == "" {
-			return reflect.Zero(f.Type), nil
+		fieldIdx, ok := fieldIndexByKey[key]
+		if !ok {
+			errs = append(errs, fmt.Errorf("unrecognized option %q", key))
+			continue
 		}
 
-		return reflect.ValueOf(strings.Split(val, ",")), nil
-	case reflect.TypeFor[map[string]bool]():
-		if val == "" {
-			return reflect.Zero(f.Type), nil
+		field := opts.Field(fieldIdx)
+		val, err := parser.popValue(field.Type())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("while parsing option %q: %w", key, err))
+			continue
 		}
+		field.Set(val)
+	}
 
-		sp := strings.Split(val, ",")
-		m := make(map[string]bool)
-		for _, s := range sp {
-			m[s] = true
-		}
-		return reflect.ValueOf(m), nil
-	case reflect.TypeFor[int]():
-		if val == "" {
-			return reflect.Zero(f.Type), nil
-		}
+	if ret.SkipLines < 0 {
```

Comment:
Suggestion: might be worth moving the validation to a separate helper function than the parsing? I can see it growing and it's semantically a bit different from parsing anyway.

Reply:
Done